### DISCUSSION TITLE
How to calculate by only CPU

### DISCUSSION THREAD

**Discussion by Yujif1Aero** [ProjectPhysX/FluidX3D#50] [Q&A] [answered]
Hi @ProjectPhysX ,@FluidX3D
,
I would like to calculate your CFD by only CPU.
Could you tell me how to do it?

Best,

**Comment by ProjectPhysX** [comment, marked as answer]
Install the [Intel OpenCL CPU runtime](https://github.com/intel/llvm/releases/tag/2022-09) (Windows: [win-oclcpuexp-2022.14.8.0.04_rel.zip](https://github.com/intel/llvm/releases/download/2022-09/win-oclcpuexp-2022.14.8.0.04_rel.zip), Linux: [oclcpuexp-2022.14.8.0.04_rel.tar.gz](https://github.com/intel/llvm/releases/download/2022-09/oclcpuexp-2022.14.8.0.04_rel.tar.gz) & [install guide](https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#install-low-level-runtime)). This works for both Intel and AMD CPUs. Once installed, your CPU will show up as an OpenCL device, and you can select it in the command line with either `./make.sh 1` or `bin'/FluidX3D.exe 1`, where the "1" is the device ID of the CPU (might be a different number for you).

Performance will be significantly slower, but the benefit is that you likely have more RAM, so you can do larger grid resolution.
